setup.py

```diff
@@ -62,9 +63,10 @@
         "evaluate",
         "protobuf>=3.20.1",
         "accelerate",  # ORTTrainer requires it.
+        "transformers<4.46.0",
     ],
-    "exporters": ["onnx", "onnxruntime", "timm"],
-    "exporters-gpu": ["onnx", "onnxruntime-gpu", "timm"],
+    "exporters": ["onnx", "onnxruntime", "timm", "transformers<4.46.0"],
+    "exporters-gpu": ["onnx", "onnxruntime-gpu", "timm", "transformers<4.46.0"],
     "exporters-tf": [
         "tensorflow>=2.4,<=2.12.1",
         "tf2onnx",
```

Comment:
Maybe we can also lift the explicit restrictions here also, since `optimum-neuron` will enforce a pinned version anyway (cc @JingyaHuang ):

```python
    "neuron": ["optimum-neuron[neuron]>=0.0.20", "transformers>=4.36.2,<4.42.0"],
    "neuronx": ["optimum-neuron[neuronx]>=0.0.20", "transformers>=4.36.2,<4.42.0"],
```

Reply:
Not related to this PR specifically so merging, let me know if you'd like me to include this change in today's release!

Reply:
@dacorvo you mean something like:
```
    "neuron": ["optimum-neuron[neuron]>=0.0.20", "transformers>=4.36.2,<=4.43.2"],
    "neuronx": ["optimum-neuron[neuronx]>=0.0.20", "transformers>=4.36.2,<=4.43.2"],
``` 
? Or you would that we remove transformers from the extra? if so wouldn't we install once again a super old optimum-neuron if we do
```
pip install optimum[neuronx]
```

Reply:
merging this PR as not directly related to this discussion, let me know if you want me to include any modification related to neuron in todays release!